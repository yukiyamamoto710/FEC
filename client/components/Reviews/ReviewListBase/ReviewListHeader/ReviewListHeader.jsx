import React from 'react';
import PropTypes from 'prop-types';

const ReviewListHeader = (props) => {
  const { len, sortBy } = props;

  const handleChangeSelect = (event) => sortBy(event.target.value);

  return (
    <div
      className="reviewListHeader"
    >
      {`${len} reviews, sorted by`}
      <select
        data-testid="select"
        className="reviewListHeader reviewListSort"
        onChange={handleChangeSelect}
      >
        <option data-testid="option">relevant</option>
        <option data-testid="option">newest</option>
        <option data-testid="option">helpful</option>
      </select>
    </div>
  );
};

ReviewListHeader.propTypes = {
  len: PropTypes.number,
  sortBy: PropTypes.func,
};

ReviewListHeader.defaultProps = {
  len: 0,
  sortBy: () => 1,
};
export default ReviewListHeader;
