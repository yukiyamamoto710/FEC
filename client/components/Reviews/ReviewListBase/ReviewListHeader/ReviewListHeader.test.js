/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  screen,
  fireEvent,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import ReviewListHeader from './ReviewListHeader';

afterEach(cleanup);

it('should render', () => {
  const testfunc = jest.fn();
  render(<ReviewListHeader len={251} sortBy={testfunc} />);
  const test = screen.queryByText('251 reviews, sorted by');
  expect(test).toBeInTheDocument();
});

it('should have default sortBy', () => {
  expect(ReviewListHeader.defaultProps.sortBy()).toBeDefined();
});

it('select should get correct option', () => {
  const testfunc = jest.fn();
  const { getByTestId, getAllByTestId } = render(<ReviewListHeader len={251} sortBy={testfunc} />);
  fireEvent.change(getByTestId('select'), { target: { value: 'newest' } });
  const options = getAllByTestId('option');
  expect(options[0].selected).toBeFalsy();
  expect(options[1].selected).toBeTruthy();
  expect(options[2].selected).toBeFalsy();
});

it('matches snapShot', () => {
  const testfunc = jest.fn();
  const tree = renderer.create(
    <ReviewListHeader
      len={251}
      sortBy={testfunc}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
