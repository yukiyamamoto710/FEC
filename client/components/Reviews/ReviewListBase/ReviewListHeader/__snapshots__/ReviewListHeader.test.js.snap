// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="reviewListHeader"
>
  251 reviews, sorted by
  <select
    className="reviewListHeader reviewListSort"
    data-testid="select"
    onChange={[Function]}
  >
    <option
      data-testid="option"
    >
      relevant
    </option>
    <option
      data-testid="option"
    >
      newest
    </option>
    <option
      data-testid="option"
    >
      helpful
    </option>
  </select>
</div>
`;
